src/stan_math_backend/Transform_Mir.ml

```diff
@@ -28,35 +28,29 @@ let rec switch_expr_to_opencl available_cl_vars e =
     | Var s when is_avail s -> {e with expr= Var (s ^ opencl_suffix)}
     | _ -> to_matrix_cl e
   in
-  let move_cl_args index arg cl_args =
+  let move_cl_args cl_args index arg =
     if List.mem ~equal:( = ) cl_args index then to_cl arg else arg
   in
-  let check_type (i, t) args = (List.nth_exn args i).emeta.mtype = t in
-  let check_if_data ind args =
+  let check_type args (i, t) = (List.nth_exn args i).emeta.mtype = t in
+  let check_if_data args ind =
     match (List.nth_exn args ind).expr with
     | Var s when is_avail s -> true
     | _ -> false
   in
-  let req_met (data_arg, type_arg) args =
-    List.for_all ~f:(fun x -> check_if_data x args) data_arg
-    && List.for_all ~f:(fun x -> check_type x args) type_arg
+  let req_met args (data_arg, type_arg) =
+    List.for_all ~f:(check_if_data args) data_arg
+    && List.for_all ~f:(check_type args) type_arg
   in
-  let any_req_met args req_args =
-    List.exists ~f:(fun x -> req_met x args) req_args
-  in
-  let get_trigger f = Map.find_exn opencl_triggers f in
-  let first (a, _) = a in
-  let second (_, a) = a in
-  let cl_args f = first (get_trigger f) in
-  let req_args f = second (get_trigger f) in
-  let map_args args f =
-    if any_req_met args (req_args f) then
-      List.mapi args ~f:(fun i a -> move_cl_args i a (cl_args f))
-    else args
+  let any_req_met args req_args = List.exists ~f:(req_met args) req_args in
+  let maybe_map_args args (cl_args, req_args) =
+    match any_req_met args req_args with
+    | true -> List.mapi args ~f:(move_cl_args cl_args)
```

Comment:
This one is supposed to be faster and also more ocamilsh right?

Reply:
Like using pattern matching here? It just seemed a little clearer to me, but it's possible it's not. I don't feel strongly about it. I'm not sure if it's faster, haha.

Reply:
I think I remember it from somewhere, idk. But yeah I agree its nicer style. Thanks.